File: src/main/java/modifiers/DataCollector.java
package modifiers;

import database.DataContainer;
import words.IWord;

import java.util.ArrayList;
import java.util.HashSet;

public class DataCollector {
    private DataContainer dataContainer;
    private ArrayList<HashSet<Integer>> wordSearchResult;

    public void setDataContainer(DataContainer dataContainer) {
        this.dataContainer = dataContainer;
    }

    public ArrayList<HashSet<Integer>> collectData(ArrayList<IWord> words) {
        for (IWord word : words) {
            searchWordInDatabase(word);
        }
        return wordSearchResult;
    }

    private void searchWordInDatabase(IWord word) {
        String wordAsString = word.getWordInString();
        HashSet<Integer> searchResult = dataContainer.getFilesNameWithSearchedWord(wordAsString);
        if (searchResult == null) searchResult = new HashSet<>();
        wordSearchResult.add(searchResult);
    }

}

File: CSharp/InvertedIndexSearcher/modifiers/dataCollector/DataCollector.cs
using System.Collections.Generic;
using InvertedIndexSearcher.database;
using InvertedIndexSearcher.words;

namespace InvertedIndexSearcher.modifiers.dataCollector
{
    public class DataCollector : IDataCollector
    {
        private readonly IDataContainer<string, HashSet<int>> _dataContainer;

        public DataCollector(IDataContainer<string, HashSet<int>> dataContainer)
        {
            _dataContainer = dataContainer;
        }

        public List<HashSet<int>> Collect(IEnumerable<IWord> words)
        {
            var listOfSearchedResultOfWords = new List<HashSet<int>>();
            foreach (var word in words)
            {
                if (!_dataContainer.GetAllData().TryGetValue(word.WordAsString, out var result))
                    result = new HashSet<int>();

                listOfSearchedResultOfWords.Add(result);
            }

            return listOfSearchedResultOfWords;
        }
    }
}